mbed_greentea/mbed_greentea_cli.py

```diff
@@ -726,140 +645,116 @@ def main_cli(opts, args, gt_instance_uuid=None):
-                    test = {"test_bin":test_bin, "image_path":image_path}
-                    test_queue.put(test)
-
-                #for test_bin, image_path in filtered_ctest_test_list.iteritems():
-                #    test = {"test_bin":test_bin, "image_path":image_path}
-                #    test_queue.put(test)
-
-                number_of_threads = 0
-                for mut in muts_to_test:
-                    #################################################################
-                    # Experimental, parallel test execution
-                    #################################################################
-                    if number_of_threads < parallel_test_exec:
-                        args = (test_result_queue, test_queue, opts, mut, mut_info, yotta_target_name, greentea_hooks)
-                        t = Thread(target=run_test_thread, args=args)
-                        execute_threads.append(t)
-                        number_of_threads += 1
+            test_list = test_build.get_tests()
+            filtered_ctest_test_list = create_filtered_test_list(test_list, opts.test_by_names, opts.skip_test)
+
+            gt_logger.gt_log("running %d test%s for platform '%s' and toolchain '%s'"% (
+                len(filtered_ctest_test_list),
+                "s" if len(filtered_ctest_test_list) != 1 else "",
+                gt_logger.gt_bright(platform_name),
+                gt_logger.gt_bright(test_build.get_toolchain())
+            ))
+
+            # Test execution order can be shuffled (also with provided random seed)
+            # for test execution reproduction.
+            filtered_ctest_test_list_keys = filtered_ctest_test_list.keys()
+            if opts.shuffle_test_order:
+                # We want to shuffle test names randomly
+                random.shuffle(filtered_ctest_test_list_keys, lambda: shuffle_random_seed)
+
+            for test_name in filtered_ctest_test_list_keys:
+                image_path = filtered_ctest_test_list[test_name].get_binary('usb').get_path()
+                test = {"test_bin":test_name, "image_path":image_path}
+                test_queue.put(test)
+
+            #for test_bin, image_path in filtered_ctest_test_list.iteritems():
+            #    test = {"test_bin":test_bin, "image_path":image_path}
+            #    test_queue.put(test)
+
+            number_of_threads = 0
+            for mut in muts_to_test:
+                #################################################################
+                # Experimental, parallel test execution
+                #################################################################
+                if number_of_threads < parallel_test_exec:
+                    args = (test_result_queue, test_queue, opts, mut, build, build_path, greentea_hooks)
+                    t = Thread(target=run_test_thread, args=args)
+                    execute_threads.append(t)
```

Comment:
No as it defaults to 1. And there is code at line 632 to validate that it is not < 1. ```number_of_threads``` is initialized to 0.